### PR TITLE
Validate account logic schema - Closes #1210

### LINKED ISSUE #1210: Account logic objectNormalize bug
### Expected Behavior

When validating account schema, it should return error or account object.

### Actual behavior

TypeError: Cannot read property '__$validated' of undefined

### Step to reproduce

Run logic account `objectNormalize` test.

### Steps to resolve

Update format to define validation rules for the account object.

Related: #677

### PR DESCRIPTION
**Pull request by robladbrook**
Closes #1210 

I'm not sure if using `anyOf` introduces any side effects for any other areas of the code using model.filter? As far as I can tell it should be fine.

Note: there is a flakey test on test/unit/logic/account.js which I think was introduced as part of #1211 
`should throw error when a numeric field receives non numeric value`

### CODE REVIEW

**Comment by LucasIsasmendi** [comment]
Hi @robladbrook, thanks for your commits. 
The way `objectNormalize` and schemas are been use in logic components is by defining an `X.property.schema` with properties and required fields and use it as second parameter when call `library.schema.validate()` inside of objectNormalize.
Also please note that we are not using es6 yet.

**Comment by robladbrook** [comment]
@LucasIsasmendi understood. What changes would you like me to make given that the test currently passes?

**Comment by LucasIsasmendi** [comment]
Instead of modify `this.model`, create `Account.prototype.schema` and use it for validations: 
```js
var report = this.scope.schema.validate(account, Account.prototype.schema);
```
Check `Transaction.prototype.schema` from `logic/transaction.js` as example

**Comment by robladbrook** [comment]
@LucasIsasmendi do these changes align with what you were thinking?

File: logic/account.js [hunk 1]
```diff
@@ -436,6 +314,188 @@ function Account (db, schema, logger, cb) {
...
+					minLength: 1
+				},
+				{
+					type: 'null'
+				}
+			]
+		},
+		address: {
+			type: 'string',
+			case: 'upper',
```
Comment (LucasIsasmendi) [inline logic/account.js]: replace case: 'upper' with `format: 'address'`

File: logic/account.js [hunk 2]
```diff
@@ -436,6 +314,188 @@ function Account (db, schema, logger, cb) {
...
+			minLength: 1,
+			maxLength: 22
+		},
+		publicKey: {
+			type: 'string',
+			format: 'publicKey'
+		},
+		secondPublicKey: {
```
Comment (LucasIsasmendi) [inline logic/account.js]: `secondPublicKey` should be the same definition as `publicKey`

File: logic/account.js [hunk 3]
```diff
@@ -436,6 +314,188 @@ function Account (db, schema, logger, cb) {
...
+				{
+					type: 'array',
+					uniqueItems: true
+				},
+				{
+					type: 'null'
+				}
+			]
+		},
+		multisignatures: {
```
Comment (LucasIsasmendi) [inline logic/account.js]: Change related multisig fields to be same as `Multisignature.prototype.schema`
- multisignatures, u_multisignatures is `keygroup`
- multimin, u_ multimin is `min`
- multilifetime, u_ multilifetime is `lifetime`

File: logic/account.js [hunk 4]
```diff
@@ -436,6 +314,188 @@ function Account (db, schema, logger, cb) {
...
+		nameexist: {
+			type: 'integer',
+			maximum: 32767
+		},
+		u_nameexist: {
+			type: 'integer',
+			maximum: 32767
+		},
+		fees: {
+			type: 'integer'
```
Comment (LucasIsasmendi) [inline logic/account.js]: Add `minimum: 0`

File: logic/account.js [hunk 5]
```diff
@@ -436,6 +314,188 @@ function Account (db, schema, logger, cb) {
...
+		},
+		rank: {
+			type: 'integer'
+		},
+		rewards: {
+			type: 'integer'
```
Comment (LucasIsasmendi) [inline logic/account.js]: Add `minimum: 0`

**Comment by robladbrook** [comment]
@LucasIsasmendi I've updated the schema, but the test now fails because the model now requires 1+ in the multisig fields.

Not sure if I should be updating the test model or allowing 0?

**Comment by LucasIsasmendi** [comment]
@robladbrook please change all min values to 0 in schema definition multisig fields

**Comment by robladbrook** [comment]
@LucasIsasmendi done!

**Pull request merged**